game-core/src/main/java/games/strategy/triplea/ai/pro/ProPurchaseAi.java

```diff
@@ -2551,6 +2571,10 @@ private void addUnitsToPlaceTerritory(
                 isBid)) {
 
           // Place max number of units
+          final List<Unit> constructions =
+              CollectionUtils.getMatches(unitsToPlace, Matches.unitIsConstruction());
+          unitsToPlace.removeAll(constructions);
```

Comment:
Why are constructions being removed from unitsToPlace and added to `ppt`. 

Second, would `ppt` provide better encapsulation if it provided a method to add constructions rather than exposing a mutable datastructure?

EG:
```
 ppt.getPlaceUnits().addAll(constructions);
```
to:
```
 ppt.addConstructions(constructions); // this method could assert that all units match 'unitIsconstruction'
```

Reply:
Essentially the method removes units from unitsToPlace and adds them into the ProPurchaseTerritory structure. The reason for this is that the AI decides how many units to build in a territory but then needs to map them to which factory is actually placing the (mostly only matters for sea territories).

Eventually, I do agree that trying to push this and a few other methods into ProPurchaseTerritory would be best just didn't want to tackle that in this PR. If you look at ProPurchaseTerritory its mostly just a data structure currently and was created to begin grouping some of the data fields together to avoid passing around lots of lists but hasn't really gotten much functionality added into it yet.